src/7.1_HW/Department.ts

```diff
@@ -0,0 +1,37 @@
+import {Employee, Status} from "./Employee";
+import {PreHiredEmployee} from "./PreHiredEmployee";
+
+export class Department {
+    name: string;
+    domain: string;
+    employees: Employee[];
+    budget: Budget;
+
+    constructor(name: string, domain: string, employees: Employee[], budget: Budget) {
+        this.name = name;
+        this.domain = domain;
+        this.employees = employees;
+        this.budget = budget;
+    }
+
+    calculateBalance(): number {
+        return this.budget.debit - this.budget.credit;
+    }
+
+    addEmployee(employee: Employee) {
+        this.employees.push(employee);
+        this.budget.debit -= employee.salary;
+    }
+
+    hirePreHiredEmployee(preHired: PreHiredEmployee): Employee {
```

Comment:
You're smart because you avoided a potential problem, but this problem turned out to be a goal :)

Requirements force you to implement adding Employee and PrehiredEmployee in a single method, challenging the use of a type guard.